src/MetadataUtility/Audio/Wave.cs

```diff
@@ -155,6 +155,24 @@ public static Fin<bool> IsPcmWaveFile(Stream stream)
             return format == Format.Pcm;
         }
 
+        public static Fin<bool> IsPreallocatedHeader(Stream stream)
+        {
+            var riffChunk = FindRiffChunk(stream);
+
+            // If file is longer than riff chunk realizes, return true
+            if (((Range)riffChunk).End < stream.Length)
```

Comment:
In [this](https://github.com/QutEcoacoustics/emu/blob/master/test/Fixtures/FL_BAR_LT/3.14_PreallocatedHeader/20191125_AAO/20191125T000000%2B1000_REC.flac) preallocated header fixture, the data section isn't empty. Was looking for some other indicator I can use to classify it as being a faulty file, noticed that the riff chunk size is short of the actual size. Maybe it makes more sense to rename the function to "IsFaultyFile", since that may not necessarily be indicative of a preallocated header? What do you think?

Reply:
So, for the fixture we have, it is an example of a _specific problem_ (in this case a pre-allocated file). You could make a `IsFaulyFile` method but it would probably invoke a few different detectors in the end since after all there are all sorts of problems. So yes, we want a specific check for pre-allocated header file. There are probably a few variants of the pre-allocated header problem.. i'm not sure at which stage FL overwrites the header with the actual flac header... we'll just have to evolve this as we go.

I think roughly the metrics for pre-allocated detection should be:

- it is a wave file with a .flac extension
- the riff chunk size is incorrect
- the data chunk size is incorrect
- first few samples of data are zeros (or there are no samples)
- the file is small (< 200 bytes)

And I think we should have this as a scoring system. If 3 out of 5 of any of these ring true then we classify it as a pre-allocated header.

Hopefully size of the file will be the biggest discriminator between pre-allocated header and a power failure (there are even more subtleties here we can get into later)